Makefile.js

```diff
@@ -208,6 +208,36 @@ function getFirstVersionOfFile(filePath) {
     return getTagOfFirstOccurrence(filePath);
 }
 
+/**
+ * Gets the commit that deleted a file.
+ * @param {string} filePath The path to the deleted file.
+ * @returns {string} The commit sha.
+ */
+function getCommitDeletingFile(filePath) {
+    var commits = execSilent("git rev-list HEAD -- " + filePath);
```

Comment:
I already did - see the parts below dealing with `versions`. Where previously it was just a map of rule name to added version, it now has two maps, one for added and one for removed. Both get persisted to the filesystem, and it will automatically update the old cache format (just added versions) to the new format.